src/main/scala/chiseltest/backends/SimulatorInterface.scala

```diff
@@ -1,30 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+
 package chiseltest.backends
 
-import firrtl.AnnotationSeq
+import chisel3.Data
+import firrtl.annotations.NoTargetAnnotation
+import logger.LazyLogging
+
+/** annotation to store [[SimulatorInterface]], which will be used in [[chiseltest.internal.ThreadedBackend]]. */
+case class SimulatorInterfaceAnnotation(interface: SimulatorInterface) extends NoTargetAnnotation
 
-/** API to Simulator. */
-trait SimulatorInterface {
+/** Standard API which should be implemented by a [[SimulatorInterface]], */
+trait SimulatorInterface extends LazyLogging {
 
   /** start time of this test. */
-  final val startTime: Long = System.nanoTime()
+  final private[chiseltest] lazy val startTime: Long = System.nanoTime()
 
-  /** how many nano seconds has elapsed since this test start. */
-  def elapsedNanoSeconds(): Long = System.nanoTime() - startTime
+  /** start time of this test. */
+  final private[chiseltest] lazy val endTime: Long = elapsedNanoSeconds()
```

Comment:
> Is this even accurate or functionally correct?

This will be evaluated at `finish`. so it's correct.

> This depends on `endTime` being only called at the end of a test - so at the least it's super brittle?

I think it's a internal API. I will make it more safe.

> Also should be orthogonal functionality.

will fix.